File: eMovieFinder/eMovieFinder.Model/Dtos/Requests/Director/DirectorUpdateRequest.cs
using System;

namespace eMovieFinder.Model.Dtos.Requests.Director
{
    public class DirectorUpdateRequest
    {
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public DateTime? BirthDate { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/dtos/requests/director/director_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'director_update_request.g.dart';

@JsonSerializable()
class DirectorUpdateRequest {
  final String? firstName;
  final String? lastName;
  final DateTime? birthDate;

  DirectorUpdateRequest({
    this.firstName,
    this.lastName,
    this.birthDate
  });

  factory DirectorUpdateRequest.fromJson(Map<String, dynamic> json) => _$DirectorUpdateRequestFromJson(json);
  Map<String, dynamic> toJson() => _$DirectorUpdateRequestToJson(this);
}